vision_agent_tools/models/model_registry.py

```diff
@@ -1,19 +1,63 @@
-from typing import Dict, Callable
+from typing import List, Dict, Type, Callable
+from enum import Enum
 from vision_agent_tools.shared_types import BaseMLModel
 from vision_agent_tools.models.florencev2 import Florencev2
+from vision_agent_tools.models.owlv2 import Owlv2
 
 
-MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {"florence2": Florencev2}
+class TextToObjectDetectionModel(str, Enum):
+    FLORENCE_V2 = "florencev2"
+    OWLV2 = "owlv2"
 
 
-def get_model(model_name: str) -> BaseMLModel:
+MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {
+    TextToObjectDetectionModel.FLORENCE_V2.value: Florencev2,
+    TextToObjectDetectionModel.OWLV2.value: Owlv2,
+}
+
+TASK_MODEL_MAP: Dict[Enum, List[str]] = {
+    TextToObjectDetectionModel: [model.value for model in TextToObjectDetectionModel]
+}
+
+
+def get_model(model_name: str, task: Type[Enum]) -> BaseMLModel:
     """
-    Retrieve a model from the registry based on the model name
+    Retrieve a model from the registry based on the model name and task
+
+    Args:
+        model_name (str): The name of the model to retrieve
+        task (Type[Enum]): The enum representing the valid models for a specific task
+
+    Returns:
+        BaseMLModel: An instance of the requested model
+
+    Raises:
+        ValueError: If the model is not registered or is not valid for the given task
     """
 
+    if model_name not in TASK_MODEL_MAP[task]:
+        raise ValueError(
+            f"Model '{model_name}' is not valid for the task '{task.__name__}'."
+        )
+
     model_class = MODEL_REGISTRY.get(model_name)
 
     if not model_class:
-        raise ValueError(f"Model '{model_name}' is not registered")
+        raise ValueError(
+            f"Model '{model_name}' is not registered in the model registry."
+        )
 
     return model_class()
+
+
+def list_models_and_tasks() -> Dict[str, List[str]]:
+    """
+    Lists available model names and the tasks they are associated with
+    """
+    models_and_tasks = {}
+
+    for task_enum, model_names in TASK_MODEL_MAP.items():
+        task_name = task_enum.__name__
+        models_and_tasks[task_name] = model_names
+
+    return models_and_tasks
```

Comment:
@hugohonda @camiloaz why do we need this function for? Is it for vision-agent use?

Reply:
it was used for debugging, if does not make sense I can remove it and leave only as a testing tool